packages/tupaia-web/src/utils/useHandleNoAccessError.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useLocation, useParams } from 'react-router';
+import { useModal } from '.';
+import { useUser } from '../api/queries';
+import { MODAL_ROUTES, URL_SEARCH_PARAMS } from '../constants';
+
+/**
+ * Returns a function that redirects the user to the login page or the request access modal(s). Needs to be a hook format because it uses hooks inside it.
+ */
+export const useHandleNoAccessError = (isProject: boolean) => {
```

Comment:
I think I might have said that we could re-use the logic between project and entity permission error handling but I now that I see it I'm finding it quite complex and hard to follow. I think that it would be better to just split this into 2 utils, one for handling project permission error and another for handling entity permission errors.

Reply:
I decided to go with the approach of handling the errors in their respective places, since that particular error handling won't get used outside of those files.